asdf_astropy/tags/transform/compound.py

```diff
@@ -0,0 +1,166 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+from asdf import tagged
+from asdf.tests.helpers import assert_tree_match
+from .basic import TransformConverter
+from astropy.modeling.core import Model, CompoundModel
+from astropy.modeling.models import Identity, Mapping, Const1D
+
+
+__all__ = ['CompoundConverter', 'RemapAxesConverter']
+
+
+_operator_to_tag_mapping = {
+    '+':  'add',
+    '-':  'subtract',
+    '*':  'multiply',
+    '/':  'divide',
+    '**': 'power',
+    '|':  'compose',
+    '&':  'concatenate',
+    'fix_inputs': 'fix_inputs'
+}
+
+
+_tag_to_method_mapping = {
+    'add':         '__add__',
+    'subtract':    '__sub__',
+    'multiply':    '__mul__',
+    'divide':      '__truediv__',
+    'power':       '__pow__',
+    'compose':     '__or__',
+    'concatenate': '__and__',
+    'fix_inputs':  'fix_inputs'
+}
+
+
+class CompoundConverter(TransformConverter):
+    tags = {"http://asdf-format.org/schemas/transform/add-2.0.0",
+            "http://asdf-format.org/schemas/transform/subtract-2.0.0",
+            "http://asdf-format.org/schemas/transform/multiply-2.0.0",
+            "http://asdf-format.org/schemas/transform/divide-2.0.0",
+            "http://asdf-format.org/schemas/transform/power-2.0.0",
+            "http://asdf-format.org/schemas/transform/compose-2.0.0",
+            "http://asdf-format.org/schemas/transform/concatenate-2.0.0",
+            "http://asdf-format.org/schemas/transform/fix_inputs-2.0.0",
+            "tag:stsci.edu:asdf/transform/add-1.0.0",
+            "tag:stsci.edu:asdf/transform/add-1.1.0",
+            "tag:stsci.edu:asdf/transform/add-1.2.0",
+            "tag:stsci.edu:asdf/transform/subtract-1.0.0",
+            "tag:stsci.edu:asdf/transform/subtract-1.1.0",
+            "tag:stsci.edu:asdf/transform/subtract-1.2.0",
+            "tag:stsci.edu:asdf/transform/multiply-1.0.0",
+            "tag:stsci.edu:asdf/transform/multiply-1.1.0",
+            "tag:stsci.edu:asdf/transform/multiply-1.2.0",
+            "tag:stsci.edu:asdf/transform/divide-1.0.0",
+            "tag:stsci.edu:asdf/transform/divide-1.1.0",
+            "tag:stsci.edu:asdf/transform/divide-1.2.0",
+            "tag:stsci.edu:asdf/transform/power-1.0.0",
+            "tag:stsci.edu:asdf/transform/power-1.1.0",
+            "tag:stsci.edu:asdf/transform/power-1.2.0",
+            "tag:stsci.edu:asdf/transform/compose-1.0.0",
+            "tag:stsci.edu:asdf/transform/compose-1.1.0",
+            "tag:stsci.edu:asdf/transform/compose-1.2.0",
+            "tag:stsci.edu:asdf/transform/concatenate-1.0.0",
+            "tag:stsci.edu:asdf/transform/concatenate-1.1.0",
+            "tag:stsci.edu:asdf/transform/concatenate-1.2.0",
+            "tag:stsci.edu:asdf/transform/fix_inputs-1.1.0",
+            "tag:stsci.edu:asdf/transform/fix_inputs-1.2.0",
+            }
+    types = {CompoundModel}
+    handle_dynamic_subclasses = True
```

Comment:
This attribute controls a feature that we're planning to remove (and not implement in `AsdfConverter`), so we can drop it.